src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -13,6 +13,15 @@
 
 #include <boost/function_output_iterator.hpp>
 
+// Note: LayerManager::annotationsEnabled is defined
+// at compile time, so that linker (with LTO on) is able
```

Comment:
nit: I think we just drop unused sections with compile flags `"-fdata-sections -ffunction-sections"` + linker flags `"--gc-sections"`. Has anyone tried compiling with -flto?

Reply:
afaik, lto enables cross-compile-unit optimization